### PR TITLE
fix: eager relation skips children relations

### LINKED ISSUE #5169: nested relations ignored if parent is eager
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

```ts
        const example = await Example.findOne(
          {  id  },
          { relations: ["parent", "parent.childOne", "parent.childTwo", "other"] }
        );
```

In this example, if `Example` entity's `parent` relation is eager: true, then `childOne` and `childTwo` will not be loaded as requested.

```ts
  @ManyToOne(
    () => Parent,
    (parent: Parent) => parent.example,
    {
      eager: true // disabling this and the above code will load nested relations
    }
  )
  @JoinColumn()
  parent: Parent | null;
```

### LINKED ISSUE #5684: An eager relationship skips children relationships
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[x] `mssql`
[x] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**
If a relation is eager there's no way to fetch its children.
Please check related PR and its test.
```ts
const user = await connection.getRepository(User).findOne(userAdmin.id, {
  relations: [
    "company", // <-- eager relation.
    "company.admin", // <-- can't be loaded without the fix.
    "company.staff", // <-- can't be loaded without the fix.
    "company.staff.company", // <-- can't be loaded without the fix.
    "company.staff.company.admin", // <-- can't be loaded without the fix.
  ],
});
```

### PR DESCRIPTION
**Pull request by satanTime**
Closes #5169
Closes #5684

### CODE REVIEW

**Comment by satanTime** [comment]
ISSUE #5684

**Comment by satanTime** [comment]
Hi @pleerock,

the PR has been updated.

**Comment by satanTime** [comment]
Hi @pleerock, could you review the PR?

**Comment by satanTime** [comment]
Hi @pleerock, could you take a look?

**Comment by satanTime** [comment]
Hi @pleerock, please let me know when you could review it.

**Comment by satanTime** [comment]
Hey @pleerock, ping.

**Comment by satanTime** [comment]
Hi @pleerock,

could you take a look at this PR when you have time? I think, the related issue is important.

**Comment by satanTime** [comment]
Hi @pleerock, @imnotjames,

I hope this finds you well.

Might you take a look at the PR? It is waiting for a review more than a year.

File: ormconfig.oracle.json
Comment (imnotjames) [inline ormconfig.oracle.json:1]: Can you remove this

File: docker/oracle/docker-entrypoint.sh
Comment (imnotjames) [inline docker/oracle/docker-entrypoint.sh:1]: Can you remove this & all the other docker related pieces

File: test/utils/test-utils.ts
Comment (imnotjames) [inline test/utils/test-utils.ts:175]: Can you remove this change - if you want to make it please include it in a separate PR.

**Review by imnotjames — CHANGES_REQUESTED** [review]
Please remove all the Oracle and docker changes.

File: test/utils/test-utils.ts
Comment (satanTime) [reply test/utils/test-utils.ts:175]: Hi, thanks for finding time to review the PR.

I was thinking about the same. The problem was that local testing / development for oracledb wasn't friendly and I added a separate instructions how to do it with a couple of cli commands.

I'll create a separate PR for this change with proper docs and remove it from this one.

File: ormconfig.oracle.json
Comment (satanTime) [reply ormconfig.oracle.json]: done

File: docker/oracle/docker-entrypoint.sh
Comment (satanTime) [reply docker/oracle/docker-entrypoint.sh]: done

File: test/utils/test-utils.ts
Comment (satanTime) [reply test/utils/test-utils.ts]: done

**Comment by satanTime** [comment]
Hi @imnotjames,

the PR has been updated to contain only the related changes.

File: src/find-options/FindOptionsUtils.ts
Comment (imnotjames) [inline src/find-options/FindOptionsUtils.ts:288]: Isn't this already shortened because of `DriverUtils.buildAlias`?

Comment (satanTime) [reply src/find-options/FindOptionsUtils.ts]: you are right, isn't needed anymore. I removed it.

**Review by imnotjames — APPROVED** [review]
I think we could refactor the `leftJoin` and `addSelect` to make this less of an issue but for now this makes sense.

**Comment by imnotjames** [comment]
Thanks for your contribution!

**Comment by satanTime** [comment]
Hi @imnotjames, thank you too!

It is nice to see, that there is a movement in the lib development after such a long time.
It is a cool lib, but, unfortunately, its maintenance was horrible.

**Comment by imnotjames** [comment]
> Hi @imnotjames, thank you too!
> 
> It is nice to see, that there is a movement in the lib development after such a long time.
> It is a cool lib, but, unfortunately, its maintenance was horrible.

Maintenance of such a large project is an incredibly demanding and difficult task - so I definitely wouldn't call any other work on this horrible.  There's just times where your priorities may not align with the maintainers. 😄  Like keeping food on the table.

**Pull request merged**
